backstory/src/test/java/com/google/sps/images/VisionImagesManagerTest.java

```diff
@@ -69,6 +111,58 @@ public void testCorrectRequestInput() throws IOException {
     ArgumentCaptor<List<AnnotateImageRequest>> argument = ArgumentCaptor.forClass(List.class);
     verify(mockImageAnnotatorClient).batchAnnotateImages(argument.capture());
     List<AnnotateImageRequest> requests = argument.getValue();
-    Assert.assertEquals(requests.size(), 1);
+    Assert.assertEquals(rawImageDataList.size(), requests.size());
+
+    // Check that features we want are requested
+    List<Feature> expectedFeatures = new ArrayList<Feature>();
+
+    Feature labelFeature = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
+    Feature landmarkFeature = Feature.newBuilder().setType(Feature.Type.LANDMARK_DETECTION).build();
+
+    expectedFeatures.add(labelFeature);
+    expectedFeatures.add(landmarkFeature);
+    
+    for (AnnotateImageRequest request: requests) {
+      Assert.assertEquals(expectedFeatures, request.getFeaturesList());
+    }
+  }
+
+  /** 
+   * Create a mock of the ImageAnnotatorClient that will return responses gotten from the
+   * passed-in AnnotatedImage List in place of actual network responses.
+   *
+   * @param desiredResponses a list of the responses we want as AnnotatedImages
+
+   */
+  private ImageAnnotatorClient mockImageAnnotatorClient(List<AnnotatedImage> desiredResponses) {
```

Comment:
FYI, this is fine given our scope, but I should probably point out that this is acting more like a "fake":
https://martinfowler.com/articles/mocksArentStubs.html

It's a bit odd to use mocks like this, so for future work I'd encourage you to create fake implementations (not using mocks) instead. Note that testing is encouraged for fakes too, since they have logic which cannot be easily inspected for correctness. 

This also applies to similar "mocks" set up for other interfaces.

Reply:
Thanks for pointing this out! I'll keep this in mind.